sdcm/sct_events/base.py

```diff
@@ -42,7 +44,68 @@
 LOGGER = logging.getLogger(__name__)
 
 
-class SctEventTypesRegistry(Dict[str, Type["SctEvent"]]):
+class Singleton(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
+
+
+class ContinuousEventRegistryError(BaseException):
+    pass
+
+
+class ContinuousEventsRegistry(metaclass=Singleton):
+    def __init__(self):
+        self._continuous_events: List[ContinuousEvent] = []
+
+    @property
+    def continuous_events(self):
+        return self._continuous_events
+
+    def add_event(self, event: ContinuousEvent):
+        if not issubclass(type(event), ContinuousEvent):
+            raise ContinuousEventRegistryError(f"Event: {event} is not a ContinuousEvent")
+
+        self.continuous_events.append(event)
+
+    def get_event(self) -> Optional[ContinuousEvent]:
```

Comment:
let's remove and fix the unit test accordingly

Reply:
Fixed